addons/editor-cleanup-plus/userscript.js

```diff
@@ -41,102 +41,95 @@ export default async function ({ addon, console, msg, safeMsg: m }) {
+    const maxWidths = result.maxWidths;
 
     for (const column of columns) {
-      let cursorY = 64;
+      let cursorY = gridSize / 2;
       let maxWidth = 0;
 
       for (const block of column.blocks) {
-        let extraHeight = 72;
-        let xy = block.getRelativeToSurfaceXY();
+        const xy = block.getRelativeToSurfaceXY();
         if (cursorX - xy.x !== 0 || cursorY - xy.y !== 0) {
           block.moveBy(cursorX - xy.x, cursorY - xy.y);
         }
-        let heightWidth = block.getHeightWidth();
-        cursorY += heightWidth.height + extraHeight;
+        const heightWidth = block.getHeightWidth();
+        cursorY += heightWidth.height + gridSize;
+        cursorY += gridSize - ((cursorY + gridSize / 2) % gridSize);
 
-        let maxWidthWithComments = maxWidths[block.id] || 0;
+        const maxWidthWithComments = maxWidths[block.id] || 0;
         maxWidth = Math.max(maxWidth, Math.max(heightWidth.width, maxWidthWithComments));
       }
 
-      cursorX += maxWidth + 96;
+      cursorX += maxWidth + gridSize;
+      cursorX += gridSize - ((cursorX + gridSize / 2) % gridSize);
     }
 
-    let topComments = workspace.getTopComments();
+    const topComments = workspace.getTopComments();
     for (const comment of topComments) {
       autoPositionComment(comment);
     }
 
     setTimeout(() => {
-      // Locate unused local variables...
-      let workspace = getWorkspace();
-      let map = workspace.getVariableMap();
-      let vars = map.getVariablesOfType("");
-      let unusedLocals = [];
-
-      for (const row of vars) {
-        if (row.isLocal) {
-          let usages = getVariableUsesById(row.getId(), workspace);
-          if (!usages || usages.length === 0) {
-            unusedLocals.push(row);
-          }
+      if (promptUnused) promptUnusedVariables();
+      UndoGroup.endUndoGroup(workspace);
+    }, 100);
+  };
+
+  function promptUnusedVariables() {
+    // Locate unused local variables...
+    const workspace = addon.tab.traps.getWorkspace();
+    const map = workspace.getVariableMap();
+    const vars = map.getVariablesOfType("");
+    const unusedVariables = [];
+
+    for (const row of vars) {
+      if (row.isLocal) {
+        const usages = getVariableUsesById(row.getId(), workspace);
+        if (!usages || usages.length === 0) {
+          unusedVariables.push(row);
         }
       }
+    }
 
-      if (unusedLocals.length > 0) {
-        const unusedCount = unusedLocals.length;
-        let message = msg("unused-var", {
-          count: unusedCount,
-        });
-        for (let i = 0; i < unusedLocals.length; i++) {
-          let orphan = unusedLocals[i];
-          if (i > 0) {
-            message += ", ";
-          }
-          message += orphan.name;
-        }
-        if (confirm(message)) {
-          for (const orphan of unusedLocals) {
-            workspace.deleteVariableById(orphan.getId());
-          }
+    if (unusedVariables.length > 0) {
```

Comment:
I think the name `unusedLocals` makes sense - it includes both variables and lists, but only local (For this sprite only) ones.